debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/BinlogReader.java

```diff
@@ -494,6 +495,33 @@ protected void logEvent(Event event) {
         logger.trace("Received event: {}", event);
     }
 
+    protected void onEvent(Event event) {
+        long ts = 0;
+
+        if (event.getHeader().getEventType() == EventType.HEARTBEAT) {
+            // HEARTBEAT events have no timestamp but are fired only when
+            // there is no traffic on the connection which means we are caught-up
+            // https://dev.mysql.com/doc/internals/en/heartbeat-event.html
+            metrics.setMilliSecondsBehindSource(ts);
+            return;
+        }
+
+        // MySQL has seconds resolution but mysql-binlog-connector-java returns
+        // a value in milliseconds
+        long eventTs = event.getHeader().getTimestamp();
+
+        if (eventTs == 0) {
+            // A zero-valued timestamp can also happen if case of ROTATE events
+            // in which case we keep the previous value
+            logger.trace("Received unexpected event with 0 timestamp: {}", event);
```

Comment:
It's not so unexpected then as per the comment above?

Reply:
Well yes but I don't understand why this happens. What I observed are two ROTATE events sequentially received with one of them with no timestamp (they are identical, this aside). And there are plenty of other events so maybe that's not the only case, maybe I should edit a bit the comment?

Reply:
yes +1; the note to ROTATE is more confusing than anything else then, I think.

Reply:
alright, let me modify it a bit

Reply:
I removed it.

Reply:
Thx!